domain/src/test/java/org/oppia/android/domain/classify/rules/fractioninput/FractionInputIsEquivalentToAndInSimplestFormRuleClassifierProviderTest.kt

```diff
@@ -0,0 +1,283 @@
+    InteractionObjectTestBuilder.createMixedNumber(
+      isNegative = false,
+      wholeNumber = 16,
+      numerator = 2,
+      denominator = 4
+    )
+  private val MIXED_NUMBER_VALUE_TEST_7_AND_1_OVER_7 =
+    InteractionObjectTestBuilder.createMixedNumber(
+      isNegative = false,
+      wholeNumber = 7,
+      numerator = 1,
+      denominator = 7
+    )
+  private val NEGATIVE_MIXED_NUMBER_VALUE_TEST_16_AND_1_OVER_2 =
+    InteractionObjectTestBuilder.createMixedNumber(
+      isNegative = true,
+      wholeNumber = 16,
+      numerator = 1,
+      denominator = 2
+    )
+
+  @Inject
+  internal lateinit var fractionInputIsEquivalentToAndInSimplestFormRuleClassifierProvider:
+    FractionInputIsEquivalentToAndInSimplestFormRuleClassifierProvider
+
+  private val inputIsEquivalentToAndInSimplestFormRuleClassifier by lazy {
+    fractionInputIsEquivalentToAndInSimplestFormRuleClassifierProvider.createRuleClassifier()
+  }
+
+  @Before
+  fun setUp() {
+    setUpTestApplicationComponent()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_wholeNumber5Answer_wholeNumber5Input_matches() {
+    val inputs = mapOf("f" to WHOLE_NUMBER_VALUE_TEST_5)
+    val answer = WHOLE_NUMBER_VALUE_TEST_5
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_wholeNumber5Answer_wholeNumber2Input_doesNotMatch() {
+    val inputs = mapOf("f" to WHOLE_NUMBER_VALUE_TEST_5)
+    val answer = WHOLE_NUMBER_VALUE_TEST_2
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_wholeNumber2Answer_negativeWholeNumber2Input_doesNotMatch() {
+    val inputs = mapOf("f" to NEGATIVE_WHOLE_NUMBER_VALUE_TEST_2)
+    val answer = WHOLE_NUMBER_VALUE_TEST_2
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_fraction1Over2Answer_fraction1Over2Input_matches() {
+    val inputs = mapOf("f" to FRACTION_VALUE_TEST_1_OVER_2)
+    val answer = FRACTION_VALUE_TEST_1_OVER_2
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isTrue()
+  }
+
+  @Test
+  fun testEquivalentAndSimplest_fraction4Over8Answer_fraction1Over2Input_doesNotMatch() {
+    val inputs = mapOf("f" to FRACTION_VALUE_TEST_1_OVER_2)
+    val answer = FRACTION_VALUE_TEST_4_OVER_8
+
+    val matches =
+      inputIsEquivalentToAndInSimplestFormRuleClassifier.matches(answer = answer, inputs = inputs)
+
+    assertThat(matches).isFalse()
+  }
+
+  // Even if creator does not input simplest form, learner's answer must be in simplest form
```

Comment:
Suggest moving this down to be before the assert line to justify it matches. Also, maybe add 'answer must still be in simplest form'

Reply:
Done